src/Facility.Definition/ServiceAttributeInfo.cs

```diff
@@ -20,8 +21,13 @@ public ServiceAttributeInfo(string name, IEnumerable<ServiceAttributeParameterIn
 			Parameters = parameters.ToReadOnlyList();
 			Position = position;
 
-			ServiceDefinitionUtility.ValidateName(Name, Position);
-			ServiceDefinitionUtility.ValidateNoDuplicateNames(Parameters, "attribute parameter");
+			this.Validate(validationMode);
+		}
+
+		IEnumerable<ServiceDefinitionError> IValidatable.Validate()
```

Comment:
Why implicit?

Reply:
`IValidatable` isn't public, so it's hiding the validation mechanics from the public API. It could be part of the API if that's useful.

Reply:
Ah, cool, I forgot to check whether `IValidatable` was public; I'm fine with it as-is.